### PR TITLE
Overhaul DAG chapter and friends

### LINKED ISSUE #37: Use simulations in DAG chapter
@LucyMcGowan if you have some existing examples to share please post here. If not I can make some up!

### LINKED ISSUE #60: 03.01: Visualizing Causal Assumptions
- [ ] https://lfoswald.github.io/2021-spring-stats2/materials/session-3/03-online-tutorial/#common-dag-path-structures-1
- [ ] Maybe: https://twitter.com/andrewheiss/status/1572037491687043072
- [ ] https://mobile.twitter.com/evpatora/status/1572619019060150274
- [ ] https://www.youtube.com/watch?app=desktop&v=Ts0hnNBRIWg&themeRefresh=1
- [ ] https://twitter.com/PWGTennant/status/1580583207241920513

### LINKED ISSUE #61: 03.02: DAGs in R: ggdag and dagitty
- [ ] #37
- [ ] #40

### LINKED ISSUE #62: 03.03: Common Structures of Bias
- [ ] https://journals.lww.com/epidem/Abstract/2022/09000/Toward_a_Clearer_Definition_of_Selection_Bias_When.13.aspx
- [ ] https://www.louisahsmith.com/publications/selection-mechanisms-and-their-consequences-understanding-and-addressing-selection-bias/selection-mechanisms-and-their-consequences-understanding-and-addressing-selection-bias.pdf
- [ ] https://elifesciences.org/articles/43154
- [ ] https://osf.io/79x4g/

### PR DESCRIPTION
**Pull request by malcolmbarrett**
Closes #60
Closes #61
Closes #62
Closes #37

Related but no longer addressed in this PR:
#149
#92
l2fu DAGs and sensitivity analyses related to DAGs

Also sets up some details for follow-up chapters, e.g. the next chapter on quartets, and partially addresses #199

### CODE REVIEW

File: chapters/chapter-05.qmd [hunk 1]
```diff
@@ -109,78 +64,244 @@ collider <- dagify(
...
+  ggplot(aes(x = x, y = y, xend = xend, yend = yend)) +
+  geom_dag_edges(edge_width = 1, edge_color = "grey85") + 
+  geom_dag_point(aes(color = deemphasize), show.legend = FALSE) + 
+  geom_dag_text() + 
+  facet_wrap(~ time) + 
+  theme_dag() + 
+  scale_color_manual(values = c("TRUE" = "grey85", "FALSE" = "black"))
+```
+
+Causality only goes forward. Association, however, is time-agnostic. It's just an observation about the numerical relationships between variables. When we control for the future, we run the risk of introducing bias. It's challenging to develop an intuition for this. Consider a case where `x` and `y` are the only causes of `q`, and all three variables are binary. When *either* `x` or `y` equals 1, then `m` happens. If we know `q = 1` and `x = 0` then logically it must be that `y = 1`. Thus, knowing about `q` gives us information about `y` via `x`. This is an extreme example, but it shows how this type of bias, sometimes called *collider-stratification bias* or *selection bias*, occurs: conditioning on `q` provides statistical information about `x` and `y` and distorts their relationship.
```
Comment (LucyMcGowan) [inline chapters/chapter-05.qmd:265]: Is `m` right here? Or should it be `q`?

File: chapters/chapter-05.qmd [hunk 2]
```diff
@@ -109,78 +64,244 @@ collider <- dagify(
...
+knitr::include_graphics("../images/podcast-diagram.png")
 ```
 
-The first step is to specify your DAG.
-In R, we can use the {ggdag} package for this along with the `dagify()` function.
-The `dagify()` function takes formulas, separated by commas, that specify cause and effect, with the left element of of the formula specifying the effect and the right all of the factors that cause it.
-What are all of the factors that "cause" graduate students to listen to a podcast the morning before an exam?
-What are all of the factors that could "cause" a graduate student to do well on a test?
-Let's posit some here.
+The main tool we'll use for making DAGs is ggdag. ggdag is a package that connects ggplot2, the most powerful visualization tool in R, to dagitty, an R package with sophisticated algorithms for querying DAGs.
```
Comment (LucyMcGowan) [inline chapters/chapter-05.qmd:300]: I think in other sections when we reference packages we've done something like `{ggdag}`

Comment (malcolmbarrett) [reply chapters/chapter-05.qmd:300]: If we want to do this globally, I think we could automate this styling with a combo of `renv::dependendencies()` and a Lua filter for the Quarto side

File: chapters/chapter-05.qmd (hunk 1 above)
Comment (malcolmbarrett) [reply chapters/chapter-05.qmd:265]: it should be `q`

File: chapters/chapter-05.qmd (hunk 2 above)
Comment (LucyMcGowan) [reply chapters/chapter-05.qmd:300]: oh great, yes love it. I also don't care if we do it or not I think I was doing it based on what was already done!

File: chapters/chapter-05.qmd (hunk 1 above)
Comment (LucyMcGowan) [reply chapters/chapter-05.qmd:265]: :+1: the m was here:

 and all three variables are binary. When *either* `x` or `y` equals 1, then `m` happens

**Comment by malcolmbarrett** [comment]
@LucyMcGowan @tgerke Ready for review. There are some outstanding `TODO`s here you can ignore. I'll take care of them as we're reviewing and before merging.

**Review by LucyMcGowan — APPROVED** [review]

**Pull request merged**